go/engine/cryptocurrency.go

```diff
@@ -48,13 +49,30 @@ func (e *CryptocurrencyEngine) SubConsumers() []libkb.UIConsumer {
 	return []libkb.UIConsumer{}
 }
 
+func normalizeAddress(address string) string {
+	switch strings.ToLower(address)[0:3] {
+	case "bc1", "zs1":
+		// bech32 addresses require that cases not be mixed
+		// (an error which will be caught downstream in validation),
+		// but the spec is otherwise case-insensitive. so if it's
+		// passed in as uppercase, then we downcase it right away to
+		// ensure that everything we do with the address (e.g. sign it into
+		// a sigchain link) will be consistent and checksum correctly.
+		if strings.ToUpper(address) == address {
```

Comment:
so what happens if it's mixed case? do we fail somewhere?

Reply:
yeah. it's recognized as invalid on the server and an error will bubble up.